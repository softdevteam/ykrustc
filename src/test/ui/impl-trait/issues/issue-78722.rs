// edition:2018

// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete
#![feature(impl_trait_in_bindings)]
//~^ WARN the feature `impl_trait_in_bindings` is incomplete

type F = impl core::future::Future<Output = u8>;

struct Bug {
    V1: [(); {
        fn concrete_use() -> F {
            async {}
        }
        let f: F = async { 1 };
        //~^ ERROR `async` blocks are not allowed in constants
        //~| ERROR destructors cannot be evaluated at compile-time
        1
    }],
}

fn main() {}
