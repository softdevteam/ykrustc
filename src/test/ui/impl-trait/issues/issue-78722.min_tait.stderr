warning: the feature `impl_trait_in_bindings` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/issue-78722.rs:7:12
   |
LL | #![feature(impl_trait_in_bindings)]
   |            ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63065 <https://github.com/rust-lang/rust/issues/63065> for more information

error: `async` blocks are not allowed in constants
  --> $DIR/issue-78722.rs:17:20
   |
LL |         let f: F = async { 1 };
   |                    ^^^^^^^^^^^

error[E0493]: destructors cannot be evaluated at compile-time
  --> $DIR/issue-78722.rs:17:13
   |
LL |         let f: F = async { 1 };
   |             ^ constants cannot evaluate destructors
...
LL |     }],
   |     - value is dropped here

error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0493`.
